packages/contract/src/contract-factory.ts

```diff
@@ -146,26 +155,190 @@ export default class ContractFactory {
   }
 
   /**
-   * Deploy a contract with the specified options.
+   * Takes a transaction request, estimates it and funds it.
+   *
+   * @param request - the request to fund.
+   * @param options - options for funding the request.
+   * @returns a funded transaction request.
+   */
+  private async fundTransactionRequest(
+    request: TransactionRequest,
+    options: DeployContractOptions = {}
+  ) {
+    const account = this.getAccount();
+    const { maxFee: setMaxFee } = options;
+
+    const txCost = await account.getTransactionCost(request);
+
+    if (isDefined(setMaxFee)) {
+      if (txCost.maxFee.gt(setMaxFee)) {
+        throw new FuelError(
+          ErrorCode.MAX_FEE_TOO_LOW,
+          `Max fee '${options.maxFee}' is lower than the required: '${txCost.maxFee}'.`
+        );
+      }
+    } else {
+      request.maxFee = txCost.maxFee;
+    }
+
+    await account.fund(request, txCost);
+
+    return request;
+  }
+
+  /**
+   * Deploy a contract of any length with the specified options.
    *
    * @param deployOptions - Options for deploying the contract.
    * @returns A promise that resolves to the deployed contract instance.
    */
   async deploy<TContract extends Contract = Contract>(
     deployOptions: DeployContractOptions = {}
   ): Promise<DeployContractResult<TContract>> {
-    if (this.bytecode.length > MAX_CONTRACT_SIZE) {
+    const account = this.getAccount();
+    const { consensusParameters } = account.provider.getChain();
+    const maxContractSize = consensusParameters.contractParameters.contractMaxSize.toNumber();
+
+    return this.bytecode.length > maxContractSize
+      ? this.deployContractAsBlobs(deployOptions)
+      : this.deployContract(deployOptions);
+  }
+
+  /**
+   * Deploy a contract with the specified options.
+   *
+   * @param deployOptions - Options for deploying the contract.
+   * @returns A promise that resolves to the deployed contract instance.
+   */
+  async deployContract<TContract extends Contract = Contract>(
```

Comment:
One of the breaking changes introduced in #2824 was the renaming of:
 - `deployContract()` to `deploy()`

The suffix is redundant given the class is called `ContractFactory`.

This PR reintroduces that method [name], but now it works differently. Aside from any confusion this can cause, I believe we should not reintroduce the `Contract` suffix to these method names. Did you explore other variations? Perhaps a middle ground:

 - `deploy()`
 - `deployAsXyz()`
 - `deployAsBlobs()`

Reply:
Nit - should we say `Blobs` or `Blob` (singular) as in "`Blob` TX" — a TX of the `Blob` type.

Just flushing ideas:
 - `deploy()`
 - `deployAsCreateTx()`
 - `deployAsBlobTx()`

Reply:
Agreed on singular. It's a little incorrect as we still submit a final create tx, but I think they are the best options so far.  Will give it till the morning to let others weigh in.

Reply:
> Nit - should we say `Blobs` or `Blob` (singular) as in "`Blob` TX" — a TX of the `Blob` type.
> 
> Just flushing ideas:
> 
> * `deploy()`
> * `deployAsCreateTx()`
> * `deployAsBlobTx()`

This seems more descriptive to me.

Reply:
SGTM, will change.